RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -0,0 +1,248 @@
+|:----     |:-------       |:-------------            | -------------------------------------------------------------------------------------------- |
+| Bitcoin  | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA-256 and then RIPEMD-160 to a user's SECP256k1 compressed public key |
+
+## Hash Time Lock Contract
+
+### Hash Functions
+
+This RFC specifies SHA-256 as the only value the `hash_function` parameter to `comit-rfc-003` may take if Bitcoin is used as a ledger.
+This may be expanded in subsequent RFCs.
+
+### Parameters
+
+The parameters for the Bitcoin HTLC follow [RFC003](./RFC-003-SWAP-basic.md#hash-time-lock-contract-htlc) and are described concretely in the following table:
+
+| Variable        | Description                                                                 |
+|:----------------|:----------------------------------------------------------------------------|
+| asset           | The quantity of satoshi                                                     |
+| secret_hash     | The hash of the `secret` (32 bytes for SHA-256)                             |
+| redeem_identity | The `pubkeyhash` of the redeeming party                                     |
+| refund_identity | The `pubkeyhash` of the refunding party                                     |
+| expiry          | The absolute UNIX timestamp in seconds after which the HTLC can be refunded |
+
+### Contract
+
+The HTLC is constructed by locking an output with the following script:
+
+```
+OP_IF
+    OP_SIZE 32 OP_EQUALVERIFY
+    OP_SHA256 <secret_hash> OP_EQUALVERIFY
+    OP_DUP OP_HASH160 <redeem_identity>
+OP_ELSE
+    <expiry> OP_CHECKLOCKTIMEVERIFY OP_DROP
+    OP_DUP OP_HASH160 <refund_identity>
+OP_ENDIF
+OP_EQUALVERIFY
+OP_CHECKSIG
+```
+
+As required by [RFC003](./RFC-003-SWAP-basic.md), this HTLC uses absolute time locks to check whether `expiry` has been reached.
+Specifically, `OP_CHECKLOCKTIMEVERIFY` (see [BIP65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki)) is used to compare the time in the block's header to the `expiry` in the contract.
+
+Implementations MUST consider an `expiry` value below `500000000` for a Bitcoin HTLC to be invalid due to the [`lock_time`](https://en.bitcoin.it/wiki/Protocol_documentation#tx) transaction field interpreting values below `500000000` as block heights.
+
+To compute the exact Bitcoin script bytes of the contract, implementations should use the following offset table:
+
+| Data              | Position of first byte | Position of last byte | Length (bytes) | Description                                                                |
+|:------------------|:-----------------------|:----------------------|:---------------|:---------------------------------------------------------------------------|
+| `6382012088a820`  | 0                      | 6                     | 7              | `OP_IF OP_SIZE 32 OP_EQUALVERIFY OP_SHA256` + `PUSH32` for the secret_hash |
```

Comment:
❓ should we use a consistent approach on how we write this table? I am asking because you put less info in the Ethereum one.

Reply:
I'm not sure. I originally started going into full detail on the Ethereum one too but it was too painful. Consistency isn't the most important attribute in RFCs and is certainly behind readability so I'd say we leave it like this for now. If we find it would be easier to implement if we just had blob of hex then we can do that.